### PR TITLE
Handle a missing ListView separator as an error

### LINKED ISSUE #23515: ListView.separated should warn user that separatorBuilder cannot be null
When using `ListView.separated`, if inside the `separatorBuilder` you return null or keep it empty, the `itemBuilder` builds only the first item. I think it would be helpful to warn the user that the `separatorBuilder` cannot be null to avoid any confusion.

The below will only build the first cell that reads 'ALPHA'. No other warnings indicate to the user why only the first item was built. If you uncomment the line in `separatorBuilder` that returns a new `Divider`, all of the items will be built.
```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  final listOfValues = ['ALPHA', 'BETA', 'GAMMA', 'DELTA', 'EPSILON'];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: ListView.separated(
          itemBuilder: (context, index) {
            return ListTile(
                title: Text(
              listOfValues[index],
              textAlign: TextAlign.center,
            ));
          },
          separatorBuilder: (context, index) {
//            return Divider();
          },
          itemCount: listOfValues.length),
    );
  }
}
```

img_d3efdbb899f5-1

```
☆ flutter doctor -v
[✓] Flutter (Channel master, v0.10.2-pre.53, on Mac OS X 10.13.6 17G65, locale en-US)
    • Flutter version 0.10.2-pre.53 at /Users/albertlardizabal/dev/flutter
    • Framework revision 549e8e07c6 (18 hours ago), 2018-10-24 14:41:16 -0700
    • Engine revision 2586e94122
    • Dart version 2.1.0-dev.7.1.flutter-45f9462398

[✓] Android toolchain - develop for Android devices (Android SDK 28.0.0-rc1)
    • Android SDK at /Users/albertlardizabal/Library/Android/sdk
    • Android NDK at /Users/albertlardizabal/Library/Android/sdk/ndk-bundle
    • Platform android-28, build-tools 28.0.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)
    • All Android licenses accepted.

[✓] iOS toolchain - develop for iOS devices (Xcode 10.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 10.0, Build version 10A255
    • ios-deploy 1.9.2
    • CocoaPods version 1.5.3

[✓] Android Studio (version 3.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 29.1.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] IntelliJ IDEA Community Edition (version 2018.2.5)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 29.1.3
    • Dart plugin version 182.4999
```

### PR DESCRIPTION
**Pull request by justinmc**
Closes https://github.com/flutter/flutter/issues/23515

Now a `separatorBuilder` that returns null looks like this:
screenshot_20181113-154845

### CODE REVIEW

**Comment by justinmc** [comment]
@HansMuller Let me know how this looks.  Is there anywhere else that needs this sort of error handling?  Ian [mentioned](https://github.com/flutter/flutter/issues/23515#issuecomment-438000313) updating Sliver as well?

**Comment by justinmc** [comment]
@HansMuller Updated, let me know if I covered everything.

I'm also worried about the difference between Error and Exception and making sure that my `_createErrorWidget` function handles both of them. https://github.com/flutter/flutter/pull/24312/files#diff-ae20aa2d21131b77d61a5fe34d70c3c0R1414

**Comment by justinmc** [comment]
@HansMuller One more try, this should cover everything we talked about.

File: packages/flutter/test/widgets/scroll_view_test.dart [hunk 1]
```diff
@@ -410,4 +411,172 @@ void main() {
...
+
+  testWidgets('separatorBuilder must return something', (WidgetTester tester) async {
+    final List<String> listOfValues = <String>['ALPHA', 'BETA', 'GAMMA', 'DELTA'];
+
+    Widget buildFrame(Widget firstSeparator) {
+      return MaterialApp(
+        home: Material(
+          child: ListView.separated(
+            itemBuilder: (BuildContext context, int index) {
+              return ListTile(
```
Comment (HansMuller) [inline packages/flutter/test/widgets/scroll_view_test.dart]: For tests, you don't really need to create a ListTile, just a Text widget will be OK

**Review by HansMuller — APPROVED** [review]
Just some cleanup stuff here.

**Comment by justinmc** [comment]
@HansMuller I fixed everything you mentioned in case you want to take a look again.  I'll try to get this to go green and then merge it.

**Pull request merged**